nixos/modules/image/repart-image.nix

```diff
@@ -77,4 +96,13 @@ runCommand imageFile
     --json=pretty \
     ${imageFile} \
     | tee repart-output.json
+
+  # Compression is implemented in the same derivation as opposed to in a
+  # separate derivation to allow users to save disk space. Disk images are
+  # already very space intensive so we want to allow users to mitigate this.
+  if [ ${builtins.toString compression.enable} ]; then
+    echo "Compressing image..."
+    ${compressionCommand} ${imageFile}
```

Comment:
I think we need to also compress all other `.raw` files generated when using `--split`